src/libs/onboardingSelectors.ts

```diff
@@ -15,6 +15,10 @@ function hasCompletedGuidedSetupFlowSelector(onboarding: OnyxValue<typeof ONYXKE
         return true;
     }
 
+    if (!isEmptyObject(onboarding) && onboarding?.hasCompletedGuidedSetupFlow === undefined) {
```

Comment:
@truph01 thanks! To avoid introducing this in future, can you please add unit tests to cover the `hasCompletedGuidedSetupFlowSelector` with various options  of the onboarding nvp? that is array, empty object, just signUpQualifier, only hasCompletedGuidedSetupFlow, no hasCompletedGuidedSetupFlow and selfTourViewed yes.